Core/clim-core/presentation-drag-and-drop.lisp

```diff
@@ -129,60 +125,70 @@
+                           last-presentation nil
+                           feedback-fn (feedback translator)
+                           highlight-fn (highlighting translator)))
+                 ;; Do not highlight the presentation if there is no
+                 ;; applicable translator. -- jd 2019-08-20
+                 (when dest-translator
+                   (maybe-funcall highlight-fn frame presentation window :highlight))
+                 (update-feedback-and-documentation
+                  (or dest-translator translator) last-presentation event window x y))))
       (block do-tracking
         (tracking-pointer (window :context-type `(or ,@(mapcar #'destination-type translators))
                                   ;; context-type should be T and we
                                   ;; should do closer examination
                                   ;; inside :presentation clause
                                   :highlight nil
                                   :multiple-window multiple-window)
-          (:presentation (&key presentation window event x y)
-            (let ((dest-translator (find-dest-translator presentation event window x y)))
-              (erase-old)
-              (if dest-translator
-                  (setf last-event event
-                        last-presentation presentation
-                        feedback-fn (feedback dest-translator)
-                        highlight-fn (highlighting dest-translator))
-                  (setf last-event event
-                        last-presentation nil
-                        feedback-fn (feedback translator)
-                        highlight-fn (highlighting translator)))
-              ;; Do not highlight the presentation if there is no
-              ;; applicable translator. -- jd 2019-08-20
-              (when dest-translator
-                (maybe-funcall highlight-fn frame presentation window :highlight))
-              (maybe-funcall feedback-fn frame from-presentation
-                             window initial-x initial-y x y :highlight)
-              (document-drag-n-drop (or dest-translator translator) last-presentation
-                                    context-type frame event window x y)))
+          (:presentation (&key presentation event window x y)
+            (update-for-presentation presentation event window x y))
           (:pointer-motion (&key event window x y)
             (erase-old)
             (setq last-event event
                   last-presentation nil)
-            (maybe-funcall feedback-fn frame from-presentation
-                           window initial-x initial-y x y :highlight)
-            (document-drag-n-drop translator nil
-                                  context-type frame event window
-                                  x y))
+            (update-feedback-and-documentation translator nil event window x y))
+          (:keyboard (&key event)
+            (let ((new-event (shallow-copy-object last-event)))
```

Comment:
this lands me in the debugger when I have shift pressed when moving Trash (because we try to make a shallow-copy of NIL).

Reply:
it seems to not be easily reproducible because now when I try it I select the text via text-selection protocol

Reply:
press pointer over a file (do not move it), and after that press the modifier

Reply:
this is still a problem. why is there a shallow copy and reinitialize-instance? can't we simply use the new event in the invocation to update-for-presentation?

Reply:
Fixed by synthesizing a pointer event at the beginning of the pointer tracking loop.